src/resources/js/views/events-bar.js

```diff
@@ -0,0 +1,161 @@
+		inputKeywordWrapperFocus: '.tribe-common-c-search__input-control--keyword-focus',
+		inputLocationWrapper: '.tribe-common-c-search__input-control--location',
+		inputLocationWrapperFocus: '.tribe-common-c-search__input-control--location-focus',
+	};
+
+	/**
+	 * Toggles input class if it has a value
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object of click event
+	 *
+	 * @return {void}
+	 */
+	obj.setInputFocusClass = function( event ) {
+		var $input = event.data.target;
+		var $wrapper = event.data.wrapper
+
+		// Set the focus class if it has content.
+		$wrapper.toggleClass( event.data.inputClassFocus, '' !== $input.val().trim() );
+	};
+
+	/**
+	 * Bind events for the keyword input of the events bar
+	 *
+	 * @since TBD
+	 *
+	 * @param {jQuery} $container jQuery object of container.
+	 *
+	 * @return {void}
+	 */
+	obj.bindEventsInputKeyword = function( $container ) {
+
+		$container
+			.find( obj.selectors.inputKeywordWrapper )
+			.each( function( index, wrapper ) {
+				$( wrapper )
+					.find( obj.selectors.input ).each( function( index, input ) {
+						$( wrapper ).toggleClass( obj.selectors.inputKeywordWrapperFocus.className(), '' !== $( input ).val().trim() );
+
+						$( input ).on( 'change', {
+							target: $( this ),
+							wrapper: $( wrapper ),
+							inputClassFocus: obj.selectors.inputKeywordWrapperFocus.className()
+						}, obj.setInputFocusClass );
+				} );
+			} );
+	};
+
+	/**
+	 * Bind events for the location input of the events bar
+	 *
+	 * @since TBD
+	 *
+	 * @param {jQuery} $container jQuery object of container.
+	 *
+	 * @return {void}
+	 */
+	obj.bindEventsInputLocation = function( $container ) {
+
+		// Bind event for the location input.
+		$container
+			.find( obj.selectors.inputLocationdWrapper )
+			.each( function( index, wrapper ) {
+				$( wrapper )
+					.find( obj.selectors.input ).each( function( index, input ) {
+						$( wrapper ).toggleClass( obj.selectors.inputLocationWrapperFocus.className(), '' !== $( input ).val().trim() );
+
+						$( input ).on( 'change', {
+							target: $( this ),
+							wrapper: $( wrapper ),
+							inputClassFocus: obj.selectors.inputLocationWrapperFocus.className()
+						}, obj.setInputFocusClass );
+				} );
+			} );
+	};
+
+	/**
+	 * Binds events for the events bar change listeners
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object for 'afterSetup.tribeEvents' event
+	 * @param {integer} index jQuery.each index param from 'afterSetup.tribeEvents' event
+	 * @param {jQuery} $container jQuery object of view container
+	 * @param {object} data data object passed from 'afterSetup.tribeEvents' event
+	 *
+	 * @return {void}
+	 */
+	obj.bindEvents = function( event, index, $container, data ) {
+		// Bind event for the keyword input.
+		obj.bindEventsInputKeyword( $container );
+
+		// Bind event for the location input.
+		obj.bindEventsInputLocation( $container );
```

Comment:
The two comments above are especially true because location input is not on by default so this logic would need to move to Pro. Let's make a generic one.